### ISSUE TITLE
Importing waypoints from Google Maps seemingly leads to crash

### ISSUE THREAD

**Issue by kebekus** [Akaflieg-Freiburg/enroute#375]
**Describe the bug**

Enroute crashes on Android when trying to accept a waypoint share from Google Maps.

**To Reproduce**

1. Open Enroute
2. Open Google Maps
3. Select a waypoint, and share it with Enroute
4. Crash

**Expected behavior**

Expect to see the waypoint description dialog.

**Comment by kebekus** [comment]
I investigated the issue:

- This is an Android-specific issue.
- The crash happens in ShareActivity.java, where setUri() is called with a NULL Argument. The first line, 'String pth = src.getPath();' in setUri then immediately leads to a crash.
- The call is made by the method processIntent(), which receives an intent of form shown below. The intent is of type SEND and does not have an extraStream. Instead, it has a text part, which can be accessed by intent.getStringExtra(Intent.EXTRA_TEXT)); the text part contains the shortened URL of the form 'https://maps.app.goo.gl/mJw3p4KMm95ehVZY8'

```
Intent { act=android.intent.action.SEND typ=text/plain flg=0x10400000 cmp=de.akaflieg_freiburg.enroute/.MobileAdaptor clip={text/plain {T(41)}} (has extras) }
```

Things to do next:

1. Get rid of the crash
2. Accept text data, and properly parse the shortened URL provided by Google
3. Understand "geo:xxx" URLs and check if they can be used in the present context
4. Understand how things work on macOS
5. Understand how things work on iOS

**Comment by kebekus** [comment]
Fixed in current development version of EFN.

**Comment by kebekus** [comment]
Fixed in 2.30.8, which is released today.
